lib/codeowners/checker.rb

```diff
@@ -93,13 +112,36 @@ def defined_owner?(file)
       false
     end
 
+    def valid_owner?(owner)
+      return true unless @validate_owners
+
+      owners_list.include?(owner)
+    end
+
     def codeowners
       @codeowners ||= CodeOwners.new(
         FileAsArray.new(codeowners_filename),
         transformers: transformers
       )
     end
 
+    def owners_list
+      return [] unless @validate_owners
+
+      @owners_list ||=
+        if github_credentials_exist?
+          Codeowners::GithubFetcher.get_owners(ENV['GITHUB_ORGANIZATION'], ENV['GITHUB_TOKEN'])
+        else
+          FileAsArray.new(ownerslist_filename).content
+        end
+    end
+
+    def persist_owners_list!
+      owners_file = FileAsArray.new(ownerslist_filename)
+      owners_file.content = @owners_list
+      owners_file.persist!
```

Comment:
Can we extract the handling of the owners file to a separate class? This class has already enough of it's own responsibilities. 

BTW. double persist, assigning to content already persists :)